vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/textfield/TextFieldWrapTest.java

```diff
@@ -78,4 +78,79 @@ public void nonInteractableField_throwsOnSetValue() {
                 () -> tf_.setValue("fail"),
                 "Setting value to a non interactable field should fail");
     }
+
+    @Test
+    void textFieldWithValidation_doNotPreventInvalid_doNotThrow() {
+        TextField tf = new TextField();
+        // Only accept numbers
+        tf.setPattern("\\d*");
+        getCurrentView().getElement().appendChild(tf.getElement());
+
+        final TextFieldWrap<TextField, String> tf_ = wrap(tf);
+        tf_.setValue("Invalid value, but doesn't throw");
+    }
```

Comment:
Does it make sense to add an assertion on value, to ensure the give text is accepted as is?

Reply:
Added